conbench/entities/history.py

```diff
@@ -143,7 +194,61 @@ def get_history(case_id: str, context_id: str, hardware_hash: str, repo: str) ->
         history_df, include_current_commit_in_rolling_stats=False
     )
 
-    return list(history_df.itertuples())
+    # return list(history_df.itertuples())
+
+    samples: list[HistorySample] = []
+    # Iterate over rows of pandas dataframe; get each row as namedtuple.
+    for sample in history_df.itertuples():
+        # Note(JP): the Commit.timestamp is nullable, i.e. not all Commit
+        # entities in the DB have a timestamp (authoring time) attached.
+        # However, in this function I believe there is an invariant that the
+        # query only returns Commits that we have this metadata for (what's the
+        # precise reason for this invariant? I think there is one). Codify this
+        # invariant with an assertion.
```

Comment:
@austin3dickey what's your take on this? 

Do we currently have commits in the database that have `timestamp` set to `None`?

If yes: is there a chance that one of the history items here is associated with one of those, or not (why?)?

Do we want / need the ability to keep track of a commit in the database that has _no_ authoring date?